File: 프로그래머스/lv0/120866. 안전지대/안전지대.py
def solution(board):
    # 방향배열
    dr = [-1, 1, 0, 0, -1, -1, 1, 1]
    dc = [0, 0, -1, 1, -1, 1, -1, 1]
    
    # board가 n*n이라는 조건
    n = len(board)
    
    # 지뢰 확인 for문
    for i in range(n):
        for j in range(n):
            if board[i][j] == 1:
            # 위험 지역 추가
                for k in range(8):
                    nr = i + dr[k]
                    nc = j + dc[k]   
                    # 위험지역은 2로 표시
                    if 0 <= nr < n and 0 <= nc < n:
                        if board[nr][nc] != 1:
                            board[nr][nc] = 2

    # 지뢰와 위험지역은 1, 2이므로 0만 카운팅
    answer = 0
    for i in range(n):
        for j in range(n):
            if board[i][j] == 0:
                answer += 1
    return answer

File: 프로그래머스/0/120866. 안전지대/안전지대.java
import java.util.*;
class Solution {
    public int solution(int[][] board) {
        int[] dr = {-1, -1, 0, 1, 1, 1, 0, -1};
        int[] dc = {0, 1, 1, 1, 0, -1, -1, -1};
        
        int n = board.length;
        int answer = n*n;
        int cnt = 0;
        
        for(int r = 0; r < n; r++){
            for(int c = 0; c < n; c++){
                if(board[r][c] == 1){
                    cnt ++;
                    for(int i = 0; i < 8; i++){
                        int nr = r + dr[i];
                        int nc = c + dc[i];
                        if(nr >= 0 && nr < n && nc >= 0 && nc < n && board[nr][nc] == 0){
                            cnt ++;
                            board[nr][nc] = 2;
                        }
                    }
                }
                
            }
        }
        return answer-cnt;
    }
}